conda_recipes/conda_build_linux_package/scripts/enter-package-build-env.sh

```diff
@@ -29,7 +29,13 @@ function conda_clean_on_error {
     if [ ! "$1" = "0" ]; then
         echo "Error detected, removing the $ENV_NAME environment and cleaning the Conda cache."
         for ENVS_DIR in $(conda info --json | python -c "import json, sys; v = json.load(sys.stdin); print('\n'.join(v['envs_dirs']))"); do
-            rm -rf $ENVS_DIR/$ENV_NAME
+            if [ -d $ENVS_DIR/$ENV_NAME ]; then
+                echo "Removing directory $ENVS_DIR/$ENV_NAME for the environment"
+                rm -rf $ENVS_DIR/$ENV_NAME
+                if [ -d $ENVS_DIR/$ENV_NAME ]; then
+                    echo "WARNING: Could not remove the directory. Possibly a permissions error or a process holding a lock."
+                fi
```

Comment:
Q: Are we missing an `exit 1` here, or is this intended?

Reply:
This is intended, so that the rest of the cleanup code runs as well.